plugins/discourse-narrative-bot/plugin.rb

```diff
@@ -170,11 +170,14 @@ module ::DiscourseNarrativeBot
 
   self.add_model_callback(Bookmark, :after_commit, on: :create) do
     if self.user.enqueue_narrative_bot_job?
-      if self.bookmarkable_type == "Post"
+      if self.bookmarkable_type == "Post" || self.bookmarkable_type == "Topic"
+        is_topic = self.bookmarkable_type == "Topic"
+        first_post_id = Post.where(topic_id: self.bookmarkable_id).first&.id if is_topic
```

Comment:
There's no need to load the whole `Post` record (done by the `.first`) just to get the `id` 😉 

```suggestion
        first_post_id = Post.where(topic_id: self.bookmarkable_id, post_number: 1).pick(:id) if is_topic
```